consensus/spos/bls/subroundEndRound.go

```diff
@@ -281,6 +306,77 @@ func (sr *subroundEndRound) doEndRoundJobByLeader() bool {
 	return true
 }
 
+// TODO:
+//	- slashing for invalid sig shares
+//	- handle sig share verifications concurrently
+func (sr *subroundEndRound) verifyNodesOnAggSigVerificationFail(
+	multiSigner crypto.MultiSigner,
+) error {
+	pubKeys := sr.ConsensusGroup()
+
+	for i, pk := range pubKeys {
+		isJobDone, err := sr.JobDone(pk, SrSignature)
+		if err != nil || !isJobDone {
+			continue
+		}
+
+		sigShare, err := multiSigner.SignatureShare(uint16(i))
+		if err != nil {
+			return err
+		}
+
+		isSuccessfull := true
+		err = multiSigner.VerifySignatureShare(uint16(i), sigShare, sr.GetData(), nil)
+		if err != nil {
+			isSuccessfull = false
+
+			err = sr.SetJobDone(pk, SrSignature, false)
+			if err != nil {
+				return err
+			}
+		}
+
+		log.Trace("verifyNodesOnAggSigVerificationFail: verifying signature share", "public key", pk, "is successfull", isSuccessfull)
+	}
+
+	return nil
+}
+
+func (sr *subroundEndRound) computeAggSigOnValidNodes(
+	multiSigner crypto.MultiSigner,
+) ([]byte, []byte, error) {
+	threshold := sr.Threshold(sr.Current())
+	numValidSigShares := sr.ComputeSize(SrSignature)
+
+	if numValidSigShares < threshold {
+		return nil, nil, fmt.Errorf("%w: number of valid sig shares lower than threshold, numSigShares: %d, threshold: %d",
+			spos.ErrInvalidNumSigShares, numValidSigShares, threshold)
+	}
+
+	bitmap := sr.GenerateBitmap(SrSignature)
+	err := sr.checkSignaturesValidity(bitmap)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sig, err := multiSigner.AggregateSigs(bitmap)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = multiSigner.SetAggregatedSig(sig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = multiSigner.Verify(sr.GetData(), bitmap)
```

Comment:
no need to re-verify the aggregated signature here since it should be correct this time, as only the valid sig shares have been aggregated.

The aggregated signature will be verified by every node on isBlockHeaderFinalInfoValid

Reply:
removed the check